### PR TITLE
Various Generator Changes

### PR DESCRIPTION
**Pull request by NinjolasNJM**
Updated many of the generators, particularly the ones that were added since the preview site went up. Notable changes are:

- Matched the Folds, Background and Labels images for the Golem Character Generator with the new Ones made for the Golem Generator
- Made completely new Images for the Cat and Cow Character Generators
- Gave overlay Region Toggles and Alex model Support to the Creeper, Cat, Cow, Enderman, Golem and Pig Character generators, and added separate Folds and Labels toggles to them if they were needed
- Tweaked Pig Generator, and changed Pig Character Generator to match its style and feature set

### CODE REVIEW

File: src/generators/minecraft-cat-character/MinecraftCatCharacterGenerator.res [hunk 1]
```diff
@@ -51,8 +48,46 @@ let script = () => {
...
+  let hideJacket = Generator.getBooleanInputValue("Hide Jacket")
+  let hideLeftSleeve = Generator.getBooleanInputValue("Hide Left Sleeve")
+  let hideRightSleeve = Generator.getBooleanInputValue("Hide Right Sleeve")
+  let hideLeftPant = Generator.getBooleanInputValue("Hide Left Pant")
+  let hideRightPant = Generator.getBooleanInputValue("Hide Right Pant")
+  let tailType =
+    Generator.getSelectInputValue("Tail Type")->Belt.Int.fromString->Belt.Option.getWithDefault(1)
+
+  let cycleTailTypes = t => {
+    let t = if t === 4 {
```
Comment (pixelpapercraft) [inline src/generators/minecraft-cat-character/MinecraftCatCharacterGenerator.res:62]: Nice!

No need to change, but there is a function `mod()` that helps with this kind of cycling.

```res
let cycleTailTypes = t => {
  let t = mod(t, 4) + 1
  Belt.Int.toString(t)
}
```
Also, not sure if you know about the `defineText()` function yet - if you would like to show a message, such as:

```res
Generator.defineText(
  "Click the tail to change its style. Currently using tail type " ++ Belt.Int.toString(tailType),
)
```

File: src/generators/minecraft-cat-character/MinecraftCatCharacterGenerator.res [hunk 2]
```diff
@@ -283,54 +612,170 @@ let script = () => {
...
+      let isArm = true
+      drawTail(sx, sy, isArm)
+      if !hideLeftSleeve {
+        drawTail(sx2, sy2, isArm)
+      }
+    }
+  | _ => ()
+  }
+
+  /* // Tail Overlay
```
Comment (pixelpapercraft) [inline src/generators/minecraft-cat-character/MinecraftCatCharacterGenerator.res:778]: If this code is not needed any more, let's remove it.

File: src/generators/minecraft-creeper-character/MinecraftCreeperCharacterGenerator.res [hunk 3]
```diff
@@ -35,11 +35,40 @@ let script = () => {
 
   // Define user variables
   Generator.defineBooleanInput("Show Folds", true)
-  Generator.defineBooleanInput("Show Overlay", true)
+  Generator.defineBooleanInput("Show Labels", true)
+  Generator.defineBooleanInput("Action Figure", false)
```
Comment (pixelpapercraft) [inline src/generators/minecraft-creeper-character/MinecraftCreeperCharacterGenerator.res:39]: Nice!

File: src/generators/minecraft-pig/MinecraftPigGenerator.res [hunk 4]
```diff
@@ -77,7 +77,7 @@ let textures: array<Generator.textureDef> = [
     standardHeight: 32,
   },
   {
-    id: "Pig (Vanilla) (Original)",
+    id: "Pig (Vanilla) (Programmer Art)",
```
Comment (pixelpapercraft) [inline src/generators/minecraft-pig/MinecraftPigGenerator.res:80]: Thanks for fixing this texture name. I'm not familiar with what programmer art is, so if it's not an official in-game texture then I'm happy to remove it.

File: src/generators/TextureMap.res [hunk 5]
```diff
@@ -141,7 +141,7 @@ module MinecraftCharacterLegacy = {
         front: {x: 36, y: 52, w: 3, h: 12},
         left: {x: 39, y: 52, w: 4, h: 12},
         back: {x: 43, y: 52, w: 3, h: 12},
-        top: {x: 43, y: 52, w: 3, h: 12},
+        top: {x: 36, y: 48, w: 3, h: 4},
```
Comment (pixelpapercraft) [inline src/generators/TextureMap.res:144]: Great 👍

**Comment by pixelpapercraft** [comment]
Amazing work @NinjolasNJM, looks great.

I've left a few comments, but I'm going to merge this as-is and if you would like to make further tweaks let's do them in another PR.

File: src/generators/minecraft-pig/MinecraftPigGenerator.res (hunk 4 above)
Comment (RubenVerg) [reply src/generators/minecraft-pig/MinecraftPigGenerator.res:80]: Programmer Art is the name given to the resource pack that ships with 1.14+ that contains the textures before the update.

Comment (pixelpapercraft) [reply src/generators/minecraft-pig/MinecraftPigGenerator.res:80]: Thanks @RubenVerg 👍

**Pull request merged**